int32 error_x
int32 error_y
uint32 number_to_punch
